jobbergate-api/jobbergate_api/apps/job_submissions/routers.py

```diff
@@ -186,6 +207,85 @@ async def job_script_update(job_submission_id: int, job_submission: JobSubmissio
     return job_submission_data
 
 
+# The "agent" routes are used for agents to fetch pending job submissions and update their statuses
+@router.get(
+    "/job-submissions/agent/pending",
+    description="Endpoint to list pending job_submissions for the requesting client",
+    response_model=List[PendingJobSubmission],
+)
+async def job_submissions_agent_pending(
+    token_payload: TokenPayload = Depends(guard.lockdown(Permissions.JOB_SUBMISSIONS_VIEW)),
+):
+    """
+    Get a list of pending job submissions for the cluster-agent.
+    """
+    if token_payload.client_id is None:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Access token does not contain a `client_id`. Cannot fetch pending submissions",
+        )
```

Comment:
This parameter is mandatory so I guess we could put it as required in the input model and let FastAPI handles

Reply:
There's no parameter on this endpoint...

Reply:
client ID I mean. IMO, the best approach is to make it mandatory

Reply:
We will make adjustments to this mechanism in a future task.